packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -257,7 +264,16 @@ export const usePickerState = <TInputValue, TValue, TDate>(
       open: isOpen,
       onClear: () => {
         // Reset all date in state to the empty value and close picker.
-        setDate({ value: valueManager.emptyValue, action: 'acceptAndClose' });
+        setDate({
+          value: valueManager.emptyValue,
+          action: 'acceptAndClose',
+          // force `onChange` in cases like input (value) === `Invalid date`
+          forceOnChangeCall: !valueManager.areValuesEqual(
+            utils,
+            value as any,
+            valueManager.emptyValue,
+          ),
```

Comment:
Should we do the same with `onAccept`?

https://codesandbox.io/s/summer-sound-rn9osw?file=/src/App.js

Reply:
Do you mean that clicking `ok` (accept) should also revert date to empty/null if it is invalid?

Reply:
Thanks for noting this case. 🙏 
Updated.
Although, I'm no longer happy with this `memo`ized `wrapperProps` as it is changing on every input value change, this forces the component to re-render even though it essentially **has** to re-render only on static picker open (to update the selected value).

Reply:
it there is not value selected, I assume that accepting no value means cleaning the field.

But I'm thinking a bit too much about what is logical instead of what user want. I assume nobody click on accept if they do not have selected a date